### PR TITLE
Clean product & combination command builders

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | https://github.com/PrestaShop/PrestaShop/issues/31373
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | The refacto is limited to command builders, which are the bridge between the form and the CQRS commands. So as long as you can save the product form and the combination form it means nothing has been broken. Automated tests should be enough to validate this maybe.
| Fixed ticket?     | part of https://github.com/PrestaShop/PrestaShop/issues/31373
| Related PRs       | 
| Sponsor company   | Your company or customer's name goes here (if applicable).

### CODE REVIEW

File: src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml:69]: If you keep this it will define the default service for this interface thus it would simplify many other service definitions with autowiring as ou don't need to explicitly set them
But it's not blocking

File: src/PrestaShopBundle/Resources/config/services/core/form/form_data_handler.yml
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/config/services/core/form/form_data_handler.yml]: With interface definition set this wouldn't be needed

**Review by jolelievre — DISMISSED** [review]
Thanks @zuk3975 

Just one suggestion about the combination builder interface (which is also valid for the product builder), but not blocking

**Comment by jolelievre** [comment]
Still some CI failing though, so maybe you can use the occasion to define the interface service that would simplify autowiring on a few services

File: src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml
Comment (zuk3975) [reply src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml:69]: We require same interface in so many places and it is designed to contain many usages, so I think its better we don't provide a default service for this interface, but always be explicit. (to avoid mistakes auto injecting wrong one)

Comment (jolelievre) [reply src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml:69]: Ok I understand your worry

**Review by jolelievre — DISMISSED** [review]
Thanks @zuk3975

File: src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml:41]:
```suggestion
```

Comment (lartist) [reply src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml:41]: Not blocking

**Review by lartist — DISMISSED** [review]

**Comment by zuk3975** [comment]
Automated tests
https://github.com/zuk3975/testing_pr/actions/runs/4383672193

**Review by MhiriFaten — COMMENTED** [review]
Hello @zuk3975 ,

I have checked your PR and I noticed this issue : 

1. Go to BO>Catalog>Products
2. Create a combination product (V2)
3. Combinations table> select product combinations
4. Bulk actions>Edit combinations 

:warning: See Error : we can not edit combinations **_`Unexpected token '<', "<!-- Prest"... is not valid JSON`_**

[Untitled_ Mar 13, 2023 12_00 PM.webm](https://user-images.githubusercontent.com/111756615/224684629-6e9a0ac6-1e0b-4919-ad43-7cfb5d5b59fe.webm)


Could you check it please ?

**Review by SharakPL — APPROVED** [review]

**Review by lartist — APPROVED** [review]

**Comment by lartist** [comment]
I approved.
However it's a shame you didn't change https://github.com/PrestaShop/PrestaShop/pull/31657#discussion_r1132149702 :)

**Comment by zuk3975** [comment]
> I approved. However it's a shame you didn't change [#31657 (comment)](https://github.com/PrestaShop/PrestaShop/pull/31657#discussion_r1132149702) :)

Yeah I remembered it after one approval sadly :smile:

**Review by MhiriFaten — APPROVED** [review]
Hello @zuk3975 ,

I have rechecked the PR and LGTM :rocket: 
Automated test are OK.
Si it is QA :heavy_check_mark: 

Thanks a lot !

**Pull request merged**
